webapp/app.py

```diff
@@ -1,25 +1,108 @@
-from flask import Flask, jsonify, redirect, render_template, url_for
+import logging
+
+from flask import Flask, Request, jsonify, redirect, render_template, session, url_for
+from flask_login import (
+    LoginManager,
+    UserMixin,
+    login_required,
+    logout_user,
+)
 from werkzeug.wrappers.response import Response
 
+from webapp.config import Config
 from webapp.exceptions import ECSTaskLogStreamDoesNotExistError
-from webapp.utils import get_task_status_and_logs
+from webapp.utils import get_task_status_and_logs, parse_oidc_data
 from webapp.utils.aws import ECSClient
 
+logger = logging.getLogger(__name__)
+
+
+class User(UserMixin):
+    """Class representing users of the app.
+
+    This class stores credentials (MIT ID, name, and email)
+    about logged in users.
+    """
+
+    def __init__(
+        self, id: str, name: str, email: str | None = None  # noqa: A002
```

Comment:
Is it worth considering something like `user_id` instead of `id` to avoid shadowing the python built-in `id`?

Reply:
Hmm, I guess if we're considering _not_ ignoring rule `A002`, I would propose updating to `mit_id` instead. What do you think, @ehanson8 ? 🤔

Reply:
Yeah, we shouldn't be ignoring `A002`, we can always be more specific than `id`.  I like `mit_id`!

Reply:
+1 to `mit_id`!  Feels ideal.  

I was a _little_ nervous that `flask-login's `UserMixin` requires an `id` field, given that some other `flask-login` flow is passing around identifiers (almost like it knows which field to use).

I think [this line](https://github.com/maxcountryman/flask-login/blob/main/src/flask_login/mixins.py#L25) in `flask-login` suggests that we **do need this `id` field** for it to function properly.

We could, of course, override all those related methods... but feels like a situation where it makes sense to ignore convention and linting a bit.  

Sorry for starting this thread in the first place!

Reply:
How do folks feel about overriding `UserMixin.get_id`?

```python
def get_id(self):
    try:
        return str(self.mit_id)
    except AttributeError:
        raise NotImplementedError("No `id` attribute - override `get_id`") from None
  ```

The error message seems to invite folks to override `get_id` method. That said, if MIT Data Warehouse decided to ever update this field name, we'd have to update the code in two places instead of one:
* custom `get_id` method
* [creating `User` instance](https://github.com/MITLibraries/alma-sapinvoices-ui/pull/60/files#diff-55f12a9f17b4ba61d3e37852150b7d78b8aedbe3629bc4ce8173d4b572297a54R83-R8

Given this, is everyone set for overriding this `A002`?

Reply:
That's a nice find about the `NotImplementedError`, and how it suggests this is _designed_ to be overridden.  I'm sold!  

I'm in full support of:
- changing `id` to `mit_id`
- overridding this method to treat `self.mit_id` as the unique, `User` identifier
- and then, linting concern `A002` no longer an issue

Reply:
Yes, that seems like an elegant solution for this situation

Reply:
The [`User` class has been updated](https://github.com/MITLibraries/alma-sapinvoices-ui/pull/60/files#diff-55f12a9f17b4ba61d3e37852150b7d78b8aedbe3629bc4ce8173d4b572297a54R27-R45):
* `id` attribute has been renamed to `mit_id`
* `get_id` method implemented (overriding `UserMixin.get_id`)
   **Note:** I chose to ignore ruff error code `EM101` to keep the try-except block consistent with parent method.

The [line of code instantiating the `User`](https://github.com/MITLibraries/alma-sapinvoices-ui/pull/60/files#diff-55f12a9f17b4ba61d3e37852150b7d78b8aedbe3629bc4ce8173d4b572297a54R53) has also been updated.